packages/strapi-admin/admin/src/containers/Webhooks/EditView/tests/reducer.test.js

```diff
@@ -0,0 +1,255 @@
+import { fromJS } from 'immutable';
+import reducer from '../reducer';
+
+describe('Admin | containers | Webhooks | EditView | reducer', () => {
+  const header = { key: '', value: '' };
+
+  const initialWebhook = {
+    events: [],
+    headers: [header],
+    name: '',
+    url: '',
+  };
+
+  const initialState = fromJS({
+    formErrors: {},
+    initialData: initialWebhook,
+    isTriggering: false,
+    modifiedData: initialWebhook,
+    triggerResponse: {},
+  });
+
+  it('It should add new header to modifiedData on ADD_NEW_HEADER', () => {
+    const state = initialState;
+
+    const action = {
+      type: 'ADD_NEW_HEADER',
+      keys: ['headers'],
+    };
+
+    const updatedHeaders = state
+      .getIn(['modifiedData', 'headers'])
```

Comment:
Here you two options to modify your state:
1. Using `setIn`
```
const updatedHeaders = state.setIn(['modifiedData', 'headers'], fromJS([header, header]));
```
2. Using `updateIn`

```
const updatedHeaders = state.updateIn(['modifiedData', 'headers'], list => {
   return list.push(fromJS(header));
});
```

Personally when writing tests with Immutable I often prefer to use setIn.